### ISSUE TITLE
Interaction between BatchNormalization and image with spatial dimension in Sequence Axis

### ISSUE THREAD

**Issue by delzac** [microsoft/CNTK#3751]
Hi all,

Does anyone knows how BatchNormalization will interact with when a spatial dimension of the image is in the sequence axis.

Example:

    channel = 3
    height = 32
    image_tensor = C.sequence.input_variable(shape=(channel, height)  # variable width rgb image
    hidden = BatchNormalization(map_rank=1)(image_tensor)

How will the elements in the sequence axis get normalized?

**Comment by haixpham** [comment]
To be exact, batch normalization applies to samples of (channel, height) size. BatchNorm only works on static axes AFAIK.

**Comment by delzac** [comment]
@haixpham Thanks for the reply! :)

Do you know what happens to the elements in the sequence axis then? Would it be counted as part of the minibatch?

**Comment by haixpham** [comment]
yes, it is counted as minibatch. It makes sense, though. Every sample must have the same size in order to gather their statistics (mean, variance).

**Comment by delzac** [comment]
Yea, i thought so too, but wasn't sure and i wasn't prepared to dig through the C++ source code to see the implementation.
